src/subdomains/generic/user/models/user/user.service.ts

```diff
@@ -173,6 +173,25 @@ export class UserService {
     return ipCountry;
   }
 
+  async blockUser(id: number, allUser = false): Promise<void> {
+    const mainUser = await this.userRepo.findOne({ where: { id }, relations: ['userData', 'userData.users'] });
+    if (!mainUser) throw new NotFoundException('User not found');
+
+    if (!allUser) {
+      if (mainUser.status === UserStatus.BLOCKED) throw new BadRequestException('User already blocked');
+      await this.userRepo.update(...mainUser.blockUser('Manual user block'));
+      return;
+    }
+
+    if (mainUser.userData.status === UserDataStatus.BLOCKED)
+      throw new BadRequestException('User Account already blocked');
+    await this.userDataService.blockUserData(mainUser.userData);
```

Comment:
Can line 181 be moved out of this allUser if? And line 186/187 should probably also be moved up. If user or user data are already blocked, we should do nothing?